monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,151 @@
+        if files:
+            pass
+        else:
+            raise ValueError("The there are no tracks in the tracks directory, create tracks first using the flight_data_generator module. ")
+
+    def make_json_compatible(self, struct: Any) -> Any:
+        if isinstance(struct, tuple) and hasattr(struct, '_asdict'):
+            return {k: self.make_json_compatible(v) for k, v in struct._asdict().items()}
+        elif isinstance(struct, dict):
+            return {k: self.make_json_compatible(v) for k, v in struct.items()}
+        elif isinstance(struct, str):
+            return struct
+        try:
+            return [self.make_json_compatible(v) for v in struct]
+        except TypeError:
+            return struct
+
+    def build_test_payload(self) ->List[DeliverablePayload]: 
+        ''' This is the main method to process the test configuration and build RID payload object, maxium of one flight is allocated to each USS. '''
+        
+        usses = self.test_configuration.usses # Store the USS details 
+
+        all_test_payloads = [] # This holds the data that will be deilver
+        
+        test_reference_time = arrow.get(self.test_configuration.now)
+        test_start_time = arrow.get(self.test_configuration.test_start_time)
+        test_start_offset = test_start_time.shift(minutes =1)
+        test_start_offset_isoformat = test_start_offset.isoformat()
+
+        requested_flights = [] # This objects holds the modified aircraft state data, aircraft state data is read from disk and timestamps are modified to have recent ones. 
+        for state_data_index, current_disk_rid_state_data in enumerate(self.disk_rid_state_data): 
+            uss = usses[state_data_index]  
+
+            if uss.injection_suffix:
+                suffix_paths = uss.injection_suffix.split('/')
+                try: 
+                    # get the test ID from the injection suffix, per the injection API, it should be the last element in the path 
+                    test_uuid = suffix_paths[-1]
+                    is_uuid = UUID(test_uuid, version=4)
+                except ValueError as ve: 
+                    test_id = str(uuid.uuid4())
+                    test_injection_path = '/tests/{test_id}'.format(test_id=test_id)
+                else: 
+                    test_id = test_uuid
+                    test_injection_path = uss.injection_suffix
+
+            else: 
+                test_id = str(uuid.uuid4())
+                test_injection_path = '/tests/{test_id}'.format(test_id=test_id)
+
+            
+            current_disk_rid_state_data['reference_time'] = test_reference_time.isoformat()
+            updated_timestamps_telemetry = []            
+
+            timestamp = test_start_offset.shift(seconds = 1)
+            
+            for telemetry_id, flight_telemetry in enumerate(current_disk_rid_state_data['flight_telemetry']['states']):
+
+                timestamp = timestamp.shift(seconds =1)
```

Comment:
This approach won't work if the timestamps in the aircraft state data aren't spaced at exactly 1 second following the reference time.  That happens to be how the data are generated at the moment, but there is no reason it will be that way in the future.  The code here should work given any valid aircraft state data.  [This](https://github.com/interuss/dss/pull/488#discussion_r626179544) is what we want to accomplish.

Reply:
OK This is fixed now, I will test it and submit for re-review